### PR TITLE
Improve integration with Groovy and Weld proxy objects

### LINKED ISSUE #256: Error marshalling a Groovy object
This example tries to marshal a simple Groovy object, but it raises the following exception 

Caused by: java.lang.IllegalAccessException: Class org.eclipse.yasson.internal.model.GetFromGetter can not access a member of class sun.reflect.annotation.AnnotatedTypeFactory$AnnotatedTypeBaseImpl with modifiers "public final"
	at sun.reflect.Reflection.ensureMemberAccess(Reflection.java:102)

Maybe it's better to limit the reflection to symbols added in GroovyObject subclasses...in many cases it's what we expect to marshal.

```groovy
import javax.json.bind.Jsonb
import javax.json.bind.JsonbBuilder

class Square {
    Integer size = 10;
}

class GroovyYassonBugTest {

    static void main(String[] args) {
        Square square = new Square()

        Jsonb jsonb = JsonbBuilder.create()
        String jsonString = jsonb.toJson(square);
    }
}
```

### LINKED ISSUE #281: How to exclude Weld metadata from JSON object serialization by Yasson
Details posted here:

https://stackoverflow.com/questions/56951279/how-to-exclude-weld-metadata-from-json-object-serialization-by-yasson

Here is approximate call stack:
	at org.eclipse.yasson.internal.model.GetValueCommand.getValue(GetValueCommand.java:36)
	at org.eclipse.yasson.internal.model.ReflectionPropagation.getValue(ReflectionPropagation.java:73)
	at org.eclipse.yasson.internal.model.PropertyModel.getValue(PropertyModel.java:285)
	at org.eclipse.yasson.internal.serializer.ObjectSerializer.marshallProperty(ObjectSerializer.java:80)
	at org.eclipse.yasson.internal.serializer.ObjectSerializer.serializeInternal(ObjectSerializer.java:61)
	at org.eclipse.yasson.internal.serializer.AbstractContainerSerializer.serialize(AbstractContainerSerializer.java:64)
	at org.eclipse.yasson.internal.Marshaller.serializeRoot(Marshaller.java:148)
	at org.eclipse.yasson.internal.Marshaller.marshall(Marshaller.java:76)
	at org.eclipse.yasson.internal.Marshaller.marshall(Marshaller.java:102)
	at org.eclipse.yasson.internal.JsonBinding.toJson(JsonBinding.java:153)
	at com.ibm.ws.jaxrs21.providers.json.JsonBProvider.writeTo(JsonBProvider.java:171)
	at org.apache.cxf.jaxrs.utils.JAXRSUtils$3.run(JAXRSUtils.java:1474)
	at org.apache.cxf.jaxrs.utils.JAXRSUtils$3.run(JAXRSUtils.java:1471)
	at java.security.AccessController.doPrivileged(AccessController.java:703)
	at org.apache.cxf.jaxrs.utils.JAXRSUtils.writeMessageBody(JAXRSUtils.java:1471)
	at org.apache.cxf.jaxrs.interceptor.JAXRSOutInterceptor.serializeMessage(JAXRSOutInterceptor.java:249)
	at org.apache.cxf.jaxrs.interceptor.JAXRSOutInterceptor.processResponse(JAXRSOutInterceptor.java:124)
	at org.apache.cxf.jaxrs.interceptor.JAXRSOutInterceptor.handleMessage(JAXRSOutInterceptor.java:85)
	at org.apache.cxf.phase.PhaseInterceptorChain.doIntercept(PhaseInterceptorChain.java:308)
	at org.apache.cxf.interceptor.OutgoingChainInterceptor.handleMessage(OutgoingChainInterceptor.java:90)
	at org.apache.cxf.phase.PhaseInterceptorChain.doIntercept(PhaseInterceptorChain.java:308)
	at org.apache.cxf.transport.ChainInitiationObserver.onMessage(ChainInitiationObserver.java:124)
	at org.apache.cxf.transport.http.AbstractHTTPDestination.invoke(AbstractHTTPDestination.java:275)
	at com.ibm.ws.jaxrs20.endpoint.AbstractJaxRsWebEndpoint.invoke(AbstractJaxRsWebEndpoint.java:134)

### PR DESCRIPTION
**Pull request by aguibert**
Fixes #281 
Fixes #256

### CODE REVIEW

**Comment by aguibert** [comment]
@m0mus or @Verdent can one of you please review?

File: src/main/java/org/eclipse/yasson/internal/ClassParser.java [hunk 1]
```diff
@@ -173,14 +174,33 @@ private void parseMethods(Class<?> clazz, JsonbAnnotatedElement<Class<?>> classE
...
 
             Property property = registerMethod(propertyName, method, classElement, classProperties);
         }
     }
+    
+    /**
+     * Filter out certain methods that get forcibly added to some classes.
+     * For example the public groovy.lang.MetaClass X.getMetaClass() method from Groovy classes
+     */
+    private boolean isSpecialCaseMethod(Class<?> clazz, Method m) {
```
Comment (Verdent) [inline src/main/java/org/eclipse/yasson/internal/ClassParser.java]: Copyright year should be 2019 instead of 2018.

Comment (aguibert) [reply src/main/java/org/eclipse/yasson/internal/ClassParser.java]: @Verdent fixed -- ready for re-review

Comment (aguibert) [reply src/main/java/org/eclipse/yasson/internal/ClassParser.java]: actually, just going to merge since I assume you have no further review comments

**Pull request merged**